pennylane/fermi/conversion.py

```diff
@@ -299,3 +300,263 @@ def _(fermi_operator: FermiSentence, n, ps=False, wire_map=None, tol=None):
+        else:
+            indices = np.append(
+                indices, _update_set(j - int(bin_rng / 2), int(bin_rng / 2)) + int(bin_rng / 2)
+            )
+    return indices
+
+
+def _parity_set(j, bin_rng):
+    """
+    Computes the parity set of the j-th orbital in n qubits.
+
+    Args:
+        j (int) : the orbital index
+        bin_rng (int) : Binary range for given number of qubits
+
+    Returns:
+        numpy.ndarray: Array of qubits which determine the parity of qubit j
+    """
+
+    indices = np.array([], dtype=int)
+    if bin_rng % 2 == 0:
+        if j < bin_rng / 2:
+            indices = np.append(indices, _parity_set(j, int(bin_rng / 2)))
+        else:
+            indices = np.append(
+                indices,
+                np.append(
+                    _parity_set(j - int(bin_rng / 2), int(bin_rng / 2)) + int(bin_rng / 2),
+                    int(bin_rng / 2 - 1),
+                ),
+            )
+
+    return indices
+
+
+def _flip_set(j, bin_rng):
+    """
+    Computes the flip set of the j-th orbital in n qubits.
+
+    Args:
+        j (int) : the orbital index
+        bin_rng (int) : Binary range for given number of qubits
+    Returns:
+        numpy.ndarray: Array containing information if the phase of orbital j is same as qubit j.
+    """
+    indices = np.array([])
+    if bin_rng % 2 == 0:
+        if j < bin_rng / 2:
+            indices = np.append(indices, _flip_set(j, int(bin_rng / 2)))
+        elif bin_rng / 2 <= j < bin_rng - 1:
+            indices = np.append(
+                indices, _flip_set(j - int(bin_rng / 2), int(bin_rng / 2)) + int(bin_rng / 2)
+            )
+        else:
+            indices = np.append(
+                np.append(indices, _flip_set(j - bin_rng / 2, bin_rng / 2) + bin_rng / 2),
+                bin_rng / 2 - 1,
+            )
+    return indices
+
+
+@singledispatch
+def _bravyi_kitaev_dispatch(fermi_operator, n, ps, wire_map, tol):
+    """Dispatches to appropriate function if fermi_operator is a FermiWord or FermiSentence."""
+    raise ValueError(f"fermi_operator must be a FermiWord or FermiSentence, got: {fermi_operator}")
+
+
+@_bravyi_kitaev_dispatch.register
+def _(fermi_operator: FermiWord, n, ps=False, wire_map=None, tol=None):
+    wires = list(fermi_operator.wires) or [0]
+    identity_wire = wires[0]
+
+    coeffs = {"+": -0.5j, "-": 0.5j}
+    qubit_operator = PauliSentence({PauliWord({}): 1.0})  # Identity PS to multiply PSs with
+
+    rng = 1
+    while np.power(2, rng) < n:
+        rng += 1
+    bin_rng = np.power(2, rng)
+
+    for item in fermi_operator.items():
+        (_, wire), sign = item
+        if wire >= n:
+            raise ValueError(
+                f"Can't create or annihilate a particle on qubit number {wire} for a system with only {n} qubits"
+            )
+
+        U_set = list(filter(lambda x: x < n, _update_set(wire, bin_rng)))
```

Comment:
Please consider using a compact format such as this or a similar one:

`u_set = [u for u in _update_set(wire, bin_rng) if u < n]`

Reply:
Also, why this logic is not inside `_update_set`? I was expecting `_update_set` to give the ultimate set without need for removing some items.

Reply:
Please also use small letters for the sets.

Reply:
This is outside the function as the function is being called recursively.

Reply:
You can still put it there as this?
```
indices = [u for u in indices if u < n]
return indices
```
Then you need to have `n` as an arg in `_update_set`.